test/tests/functional/pbs_ralter.py

```diff
@@ -2279,20 +2279,38 @@ def test_alter_standing_resv_future_occrs(self):
             offset, duration, select="2:ncpus=4", standing=True,
             rrule="FREQ=HOURLY;COUNT=3")
 
-        self.server.expect(RESV,
-                           {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')},
-                           rid)
         # move the reservation 10 mins in future
         self.alter_a_reservation(rid, start, end, confirm=True, alter_s=True,
                                  alter_e=True, shift=shift)
-        self.server.expect(RESV,
-                           {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')},
-                           rid)
-        # Now make another reservation which runs in about 50 mins from now.
         # Ideally this reservation should confirm because second occurrence
         # of the first reservation happens in almost 2 hrs from now.
         rid2, start, end = self.submit_and_confirm_reservation(
             3000, 1800, select="2:ncpus=4")
+
+    def test_alter_standing_resv_check_start_time(self):
+        """
+        Test that when start time of a confirmed standing reservation is
+        altered, all its occurrences start at the right time.
+        """
```

Comment:
I don't  think this test is required.  The server always did the right thing.  I don't even think you need the other test you wrote either.  If you want to write a test, then write one where the scheduler overlaps a reservation with a future occurrence because it thinks the occurrences are sooner than they are.

Reply:
In this test we are checking that server is doing the right thing by starting the reservation at the right time. @tpai26 suggested me to write this test because of the recent inflow of problems that we have seen with the ralter changes.

I think the other test does check for the reservation to be denied. Since we change the duration of standing reservation from 180 to 20 seconds, buggy behavior will confirm the second reservation.